apps/condo/domains/property/schema/Property.js

```diff
@@ -217,12 +233,24 @@ const Property = new GQLListSchema('Property', {
     hooks: {
         validateInput: ({ resolvedData, existingItem, context, addValidationError }) => {
             if (!hasDvAndSenderFields(resolvedData, context, addValidationError)) return
-            if (!hasDbFields(['organization', 'type', 'address', 'addressMeta'], resolvedData, existingItem, context, addValidationError)) return
+            if (!hasDbFields(REQUIRED_FIELDS, resolvedData, existingItem, context, addValidationError)) return
+
             const { dv } = resolvedData
+
             if (dv === 1) {
                 // NOTE: version 1 specific translations. Don't optimize this logic
             } else {
-                return addValidationError(`${DV_UNKNOWN_VERSION_ERROR}dv] Unknown \`dv\``)
+                return addValidationError(`${DV_UNKNOWN_VERSION_ERROR}dv] Unknown 'dv'`)
+            }
+        },
+        afterChange: async ({ operation, existingItem, updatedItem }) => {
+            const isSoftDeleteOperation = operation === 'update' && !existingItem.deletedAt && Boolean(updatedItem.deletedAt)
+            const isCreatedProperty = operation === 'create' && Boolean(updatedItem.address)
+            const isRestoredProperty = operation === 'update' && Boolean(existingItem.deletedAt) && !updatedItem.deletedAt
+            const id = isSoftDeleteOperation ? existingItem.id : updatedItem.id
+
+            if (isSoftDeleteOperation || isCreatedProperty || isRestoredProperty) {
```

Comment:
these are not all possible cases. please do not do unnecessary optimization if you are not sure that you have taken into account all the edge cases. It's Ok to run the task in all property change cases

Reply:
fixed